py/objobject.c

```diff
@@ -62,13 +62,51 @@ STATIC mp_obj_t object___new__(mp_obj_t cls) {
 STATIC MP_DEFINE_CONST_FUN_OBJ_1(object___new___fun_obj, object___new__);
 STATIC MP_DEFINE_CONST_STATICMETHOD_OBJ(object___new___obj, MP_ROM_PTR(&object___new___fun_obj));
 
+#if MICROPY_PY_DELATTR_SETATTR
+STATIC mp_obj_t object___setattr__(mp_obj_t self_in, mp_obj_t attr, mp_obj_t value) {
+    if (!mp_obj_is_instance_type(mp_obj_get_type(MP_OBJ_TO_PTR(self_in)))) {
+        mp_raise_TypeError("__setattr__ arg must be a user-type");
```

Comment:
Might be an idea to optimise the error messages to reduce code size.  In the extreme case could just pass NULL as the error messages.  Less extreme would be to use "arg must be user-type" here and in delattr, and "attr must be str" for the string check (or even NULL for that one since it's unlikely to be hit).

Reply:
Ok. Changed the error message in `object.__new__` so it is merged as well.